# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/DeviceManagerDx.cpp
# Commit: 7707637480e7
# Full Hash: 7707637480e7a5efa16c33cee89f10f5ce6be48f
# Author: Kelsey Gilbert <kelsey.gilbert@mozilla.com>
# Date: 2023-02-07 04:35:34
# Regressor Bug: 1799258
# File Overlap Count: 1
# Description:
#   Bug 1799258 - Share all-of-dcomp.h preamble, and deal with outdated mingw dcomp.h. r=gfx-reviewers,sotaro
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D168839
# ==============================================================================

diff -r 0141b29bf5df -r 7707637480e7 gfx/thebes/DeviceManagerDx.cpp
--- a/gfx/thebes/DeviceManagerDx.cpp	Mon Feb 06 19:58:55 2023 +0000
+++ b/gfx/thebes/DeviceManagerDx.cpp	Mon Feb 06 19:58:56 2023 +0000
@@ -25,17 +25,10 @@
 #include "nsPrintfCString.h"
 #include "nsString.h"
 
-#undef _WIN32_WINNT
-#define _WIN32_WINNT _WIN32_WINNT_WINBLUE
-#undef NTDDI_VERSION
-#define NTDDI_VERSION NTDDI_WINBLUE
+// -
 
-// We also need this, or dcomp.h won't give us e.g. IDCompositionDevice3:
-#undef _WIN32_WINNT_WINTHRESHOLD
-#define _WIN32_WINNT_WINTHRESHOLD _WIN32_WINNT
-
+#include "mozilla/gfx/AllOfDcomp.h"
 #include <d3d11.h>
-#include <dcomp.h>
 #include <ddraw.h>
 #include <dxgi.h>
 